theseus/embodied/robot/robot.py

```diff
@@ -0,0 +1,175 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import abc
+from typing import List, Dict, Optional
+import urdf_parser_py.urdf as urdf
+import torch
+
+from theseus.geometry.functional import so3, se3
+from .joint import Joint, FixedJoint, RevoluteJoint, PrismaticJoint
+from .link import Link
+
+
+class Robot(abc.ABC):
+    def __init__(self, name: str, dtype: torch.dtype = None):
+        if dtype is None:
+            dtype = torch.get_default_dtype()
+        self._name: str = name
+        self._dtype: torch.dtype = dtype
```

Comment:
Maybe can consolidate the if block and assignment line for _dtype.

Reply:
refactored.